Add firmware upgrades

philippedev101 opened the pull request:
https://github.com/haskellfoundation/hf-infrastructure/issues/21

philippedev101 committed:
Add firmware upgrades

chreekat commented on hf-cert-1/default.nix:
```diff
@@ -44,5 +44,7 @@ in
   };
   networking.firewall.allowedTCPPorts = [ 80 443 ];
 
+  services.fwupd.enable = true;
+
```
This file imports system-common! :) So I don't think we need to enable it here.

chreekat commented on stackage-builder/configuration.nix:
```diff
@@ -89,6 +89,8 @@
   # services.openssh.enable = true;
 
+  services.fwupd.enable = true;
+
```
Actually, I believe this file is unused and can just be deleted. In fact, `stackage-builder/nixos-modules` is good, but everything else under `stackage-builder` is unused after the migration to opencape!

chreekat reviewed (changes requested):
Thanks for this. Would you be willing to change things a bit in order to also improve the overall health of the repo?

1. Add the contents of nix-hygiene.nix into system-common.nix, and delete nix-hygiene.nix
2. Rename system-common.nix to shared/system-common.nix
3. Export nixosModules.system-common instead of nixosModules.nix-hygiene
4. Remove the FIXME on flake.nix line 30 while you're doing 3 :)

philippedev101 committed:
Apply review feedback and fix flake check

- Remove redundant `fwupd` setting in `hf-cert-1`.
- Delete the obsolete `stackage-builder/configuration.nix` file.
- Re-create the `stackage-builder` umbrella module in `flake.nix`.
- Pass `inputs` to the test VM so it can import `sops-nix`.

chreekat commented:
This PR was spiritually implemented by e503523f7022dc5eedc496c113a09d8592782acc . :) I had to clean up a few other things at the same time, so I didn't end up merging this as it was. Sorry about that. I added you as a co-author to that commit!
